patterns/builder.md

```diff
@@ -2,46 +2,58 @@
 
 ## Description
 
-Construct an object with calls to a builder helper. 
-
+Construct an object with calls to a builder helper.
 
 ## Example
 
 ```rust
-struct Foo {
+#[derive(Debug)]
+pub struct Foo {
     // Lots of complicated fields.
+    bar: String,
+}
+
+impl PartialEq for Foo {
+    fn eq(&self, other: &Self) -> bool {
+        self.bar == other.bar
+    }
 }
```

Comment:
Why not just do derive `PartialEq` above? Rather than writing this from scratch?